File: StanfordSoftware/Samples/StanfordParser.Fsharp.Samples/ParserDemo2.fs
module ParserDemo2

open java.io
open java.util

open edu.stanford.nlp.ling
open edu.stanford.nlp.``process``
open edu.stanford.nlp.trees
open edu.stanford.nlp.parser.lexparser

// Usage: ParserDemo2 [[grammar] textFile] 
let main (model:string option) (fileName:string option) = 
    let grammar = 
        match model with
        | Some(model) -> model
        | None -> @"..\..\..\..\temp\stanford-parser-full-2013-06-20\edu\stanford\nlp\models\lexparser\englishPCFG.ser.gz"
    let options =[|"-maxLength"; "80"; "-retainTmpSubcategories"|]
    let lp = LexicalizedParser.loadModel(grammar, options);
    let tlp = PennTreebankLanguagePack();
    let gsf = tlp.grammaticalStructureFactory();

    let sentences = 
        match fileName with
        | Some(file) -> 
            DocumentPreprocessor(file)
            |> Iterable.toSeq
            |> Seq.cast<ArrayList>
        | None ->
            // Showing tokenization and parsing in code a couple of different ways.
            seq {
                let sent = [| "This"; "is"; "an"; "easy"; "sentence"; "." |] 
                yield (sent |> Seq.map (fun w-> Word(w)) |> Iterable.fromSeq) :?> ArrayList

                let sent2 ="This is a slightly longer and more complex sentence requiring tokenization."
                let toke = tlp.getTokenizerFactory().getTokenizer(new StringReader(sent2));
                yield toke.tokenize() :?> ArrayList
            }

    for sentence in sentences do
        let parse = lp.apply(sentence)
        parse.pennPrint()
        printfn "\n%O\n" (parse.taggedYield())
        
        let gs = gsf.newGrammaticalStructure(parse)
        let tdl = gs.typedDependenciesCCprocessed(true)
        printfn "%O\n" tdl

File: StanfordSoftware/Samples/StanfordParser.Csharp.Samples/ParserDemo2.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using edu.stanford.nlp.ling;
using edu.stanford.nlp.parser.lexparser;
using edu.stanford.nlp.process;
using edu.stanford.nlp.trees;

using java.io;
using java.util;

namespace StanfordParser.Csharp.Samples
{
    public static class ParserDemo2
    {
        public static void Start(string model, string fileName)
        {
            var grammar = (!String.IsNullOrEmpty(model)) ? model : Program.ParserModel;
            var options = new[] { "-maxLength", "80", "-retainTmpSubcategories" };
            var lp = LexicalizedParser.loadModel(grammar, options);
            var tlp = new PennTreebankLanguagePack();
            var gsf = tlp.grammaticalStructureFactory();

            var sentences = new List<ArrayList>();
            if (!string.IsNullOrEmpty(fileName))
            {
                sentences.AddRange(new DocumentPreprocessor(fileName).Cast<ArrayList>());
            }
            else
            {
                var sent = new[] { "This", "is", "an", "easy", "sentence", "." };
                var arrList = new ArrayList();
                foreach (var s in sent)
                {
                    arrList.Add(new Word(s));
                }
                sentences.Add(arrList);

                const string Sent2 = "This is a slightly longer and more complex sentence requiring tokenization.";
                var toke = tlp.getTokenizerFactory().getTokenizer(new StringReader(Sent2));
                sentences.Add((ArrayList)toke.tokenize());
            }

            foreach (var sentence in sentences)
            {
                var parse = lp.apply(sentence);
                parse.pennPrint();
                System.Console.WriteLine("\n{0}\n", (parse.taggedYield()));

                var gs = gsf.newGrammaticalStructure(parse);
                var tdl = gs.typedDependenciesCCprocessed(true);
                System.Console.WriteLine("{0}\n", tdl);
            }
       
        }
    }
}
